lib/elixir/lib/calendar/iso.ex

```diff
@@ -715,6 +715,126 @@ defmodule Calendar.ISO do
     {0, 1}
   end
 
+  @doc """
+  Implements the inspect protocol for a date
+  delegated from `Date`.
+
+  """
+  @doc since: "1.10.0"
+  @impl true
+  @spec inspect_date(Calendar.year(), Calendar.month(), Calendar.day(), Inspect.Opts.t()) ::
+          String.t()
+  def inspect_date(year, month, day, _) do
+    "~D[" <> date_to_string(year, month, day) <> "]"
+  end
+
+  @doc """
+  Implements the inspect protocol for a datetime
+  delegated from `DateTime`.
+
+  """
+  @doc since: "1.10.0"
+  @impl true
+  @spec inspect_datetime(
+          Calendar.year(),
+          Calendar.month(),
+          Calendar.day(),
+          Calendar.hour(),
+          Calendar.minute(),
+          Calendar.second(),
+          Calendar.microsecond(),
+          Calendar.time_zone(),
+          Calendar.zone_abbr(),
+          Calendar.utc_offset(),
+          Calendar.std_offset(),
+          Inspect.Opts.t()
+        ) ::
+          String.t()
+  def inspect_datetime(
+        year,
+        month,
+        day,
+        hour,
+        minute,
+        second,
+        microsecond,
+        time_zone,
+        zone_abbr,
+        utc_offset,
+        std_offset,
+        _opts
+      ) do
+    formatted =
+      datetime_to_string(
+        year,
+        month,
+        day,
+        hour,
+        minute,
+        second,
+        microsecond,
+        time_zone,
+        zone_abbr,
+        utc_offset,
+        std_offset
+      )
+
+    if utc_offset == 0 && std_offset == 0 && time_zone == "Etc/UTC" do
+      "~U[" <> formatted <> "]"
+    else
+      "#DateTime<" <> formatted <> ">"
+    end
+  end
+
+  @doc """
+  Implements the inspect protocol for a naive
+  datetime delegated from `NaiveDateTime`.
+
+  """
+  @doc since: "1.10.0"
+  @impl true
+  @spec inspect_naive_datetime(
+          Calendar.year(),
+          Calendar.month(),
+          Calendar.day(),
+          Calendar.hour(),
+          Calendar.minute(),
+          Calendar.second(),
+          Calendar.microsecond(),
+          Inspect.Opts.t()
+        ) ::
+          String.t()
+  def inspect_naive_datetime(year, month, day, hour, minute, second, microsecond, _opts) do
+    formatted = naive_datetime_to_string(year, month, day, hour, minute, second, microsecond)
+
+    "~N[" <> formatted <> "]"
+  end
+
+  @doc """
+  Implements the inspect protocol for a time
+  delegated from `Time`.
+
```

Comment:
Minor nitpick: this blank line at the end of the `@doc` is not common in the Elixir codebase (unless you have indentation) 🙂